### PR TITLE
Fix part of #2294: Migrate to TextInputLayout in AdminPinActivity

### PR DESCRIPTION
**Pull request by prayutsu**
## Explanation

Fixes part of #2294, #962 
Migration for `AdminPinActivity`

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

| Portrait | Landscape |
| --------   | -----------:    | 
|Screenshot_1609331965|Screenshot_1609332070|
|Screenshot_1609331975|Screenshot_1609332078|
|Screenshot_2020-12-30-18-17-02-364_org oppia android|Screenshot_2020-12-30-18-17-47-076_org oppia android|
|Screenshot_2020-12-30-18-17-06-426_org oppia android|Screenshot_2020-12-30-18-17-25-096_org oppia android|

Video - 

https://user-images.githubusercontent.com/54636525/103354155-ca24e280-4ad0-11eb-9c86-f1c85637c233.mp4 

https://user-images.githubusercontent.com/54636525/103363233-98b41300-4ae0-11eb-9d57-f312d3bf66a3.mp4

Espresso Tests - 
Screenshot from 2021-01-01 12-13-55

Robolectric Tests - 
Screenshot from 2021-01-01 12-20-05

### CODE REVIEW

**Comment by prayutsu** [comment]
@anandwana001 PTAL

**Comment by anandwana001** [comment]
is there a dark mode in the app?

**Comment by FareesHussain** [comment]
> is there a dark mode in the app?

Some mobiles forces the app to follow system settings for theme
This same for my mobile too (Redmi note 7 pro)

File: app/src/main/res/layout/admin_pin_activity.xml [hunk 1]
```diff
@@ -57,35 +56,53 @@
...
+        app:layout_constraintTop_toBottomOf="@+id/admin_pin_warning_text">
 
-      <org.oppia.android.app.profile.ProfileInputView
+        <com.google.android.material.textfield.TextInputEditText
+          android:id="@+id/input_pin_edit_text"
+          style="@style/TextInputEditText"
+          android:layout_width="match_parent"
+          android:layout_height="wrap_content"
+          android:hint="@string/admin_pin_new_pin"
+          android:imeOptions="actionDone"
```
Comment (anandwana001) [inline app/src/main/res/layout/admin_pin_activity.xml]: this should be `actionNext` so the next input gets ready for input

File: app/src/main/res/layout/admin_pin_activity.xml [hunk 2]
```diff
@@ -57,35 +56,53 @@
         app:layout_constraintStart_toStartOf="parent"
         app:layout_constraintTop_toBottomOf="@+id/admin_pin_main_text" />
 
-      <org.oppia.android.app.profile.ProfileInputView
+      <com.google.android.material.textfield.TextInputLayout
         android:id="@+id/input_pin"
```
Comment (anandwana001) [inline app/src/main/res/layout/admin_pin_activity.xml]: update id with more understandable naming inclined towards the screen. 
for reference from other file `admin_auth_input_pin`

**Review by anandwana001 — CHANGES_REQUESTED** [review]
Thaks @prayutsu 
nit changes suggested.

File: app/src/main/res/layout/admin_pin_activity.xml (hunk 2 above)
Comment (prayutsu) [reply app/src/main/res/layout/admin_pin_activity.xml]: Does `admin_pin_input_pin` sound good?

File: app/src/main/res/layout-land/admin_pin_activity.xml [hunk 3]
```diff
@@ -57,36 +56,54 @@
...
+          android:layout_height="wrap_content"
+          android:hint="@string/admin_pin_new_pin"
+          android:imeOptions="actionDone"
+          android:inputType="numberPassword"
+          android:maxLength="5" />
+      </com.google.android.material.textfield.TextInputLayout>
+
+      <com.google.android.material.textfield.TextInputLayout
         android:id="@+id/input_confirm_pin"
+        style="@style/TextInputLayout"
```
Comment (prayutsu) [inline app/src/main/res/layout-land/admin_pin_activity.xml:87]: @anandwana001 Should I update this id too?

**Review by prayutsu — COMMENTED** [review]
Nit suggestions.

File: app/src/main/res/layout/admin_pin_activity.xml (hunk 2 above)
Comment (prayutsu) [reply app/src/main/res/layout/admin_pin_activity.xml:60]: `admin_pin_input_pin` is already taken in `pin_password_activity`. Can you please suggest some other name?

Comment (prayutsu) [reply app/src/main/res/layout/admin_pin_activity.xml]: Done

File: app/src/main/res/layout/admin_pin_activity.xml (hunk 1 above)
Comment (prayutsu) [reply app/src/main/res/layout/admin_pin_activity.xml]: Done

File: app/src/main/res/layout-land/admin_pin_activity.xml [hunk 4]
```diff
@@ -57,36 +56,54 @@
...
-        profile:error="@{viewModel.pinErrorMsg}" />
+        app:layout_constraintTop_toBottomOf="@+id/admin_pin_warning_text">
 
-      <org.oppia.android.app.profile.ProfileInputView
-        android:id="@+id/input_confirm_pin"
+        <com.google.android.material.textfield.TextInputEditText
+          android:id="@+id/admin_pin_input_pin_edit_text"
+          style="@style/TextInputEditText"
+          android:layout_width="match_parent"
+          android:layout_height="wrap_content"
```
Comment (FareesHussain) [inline app/src/main/res/layout-land/admin_pin_activity.xml:77]: Add this to support RTL for TextInputLayout
`android:textDirection="locale"`

File: app/src/main/res/layout-land/admin_pin_activity.xml [hunk 5]
```diff
@@ -57,36 +56,54 @@
...
-        app:label="@string/admin_pin_new_confirm"
+        app:errorMessage="@{viewModel.confirmPinErrorMsg}"
         app:layout_constraintEnd_toEndOf="parent"
         app:layout_constraintHorizontal_bias="0.0"
         app:layout_constraintStart_toStartOf="parent"
-        app:layout_constraintTop_toBottomOf="@+id/input_pin"
-        profile:error="@{viewModel.confirmPinErrorMsg}" />
+        app:layout_constraintTop_toBottomOf="@+id/admin_pin_input_pin">
+
+        <com.google.android.material.textfield.TextInputEditText
```
Comment (FareesHussain) [inline app/src/main/res/layout-land/admin_pin_activity.xml:100]: Add this to support RTL for TextInputLayout
`android:textDirection="locale"`

File: app/src/main/res/layout-sw600dp-land/admin_pin_activity.xml [hunk 6]
```diff
@@ -61,32 +60,50 @@
...
+          app:errorMessage="@{viewModel.pinErrorMsg}"
           app:layout_constraintEnd_toEndOf="parent"
           app:layout_constraintStart_toStartOf="parent"
-          app:layout_constraintTop_toBottomOf="@+id/admin_pin_warning_text"
-          profile:error="@{viewModel.pinErrorMsg}" />
+          app:layout_constraintTop_toBottomOf="@+id/admin_pin_warning_text">
 
-        <org.oppia.android.app.profile.ProfileInputView
-          android:id="@+id/input_confirm_pin"
+          <com.google.android.material.textfield.TextInputEditText
```
Comment (FareesHussain) [inline app/src/main/res/layout-sw600dp-land/admin_pin_activity.xml:75]: Add this to support RTL for TextInputLayout
`android:textDirection="locale"`

File: app/src/main/res/layout-sw600dp-land/admin_pin_activity.xml [hunk 7]
```diff
@@ -61,32 +60,50 @@
...
-          app:label="@string/admin_pin_new_confirm"
+          app:errorMessage="@{viewModel.confirmPinErrorMsg}"
           app:layout_constraintEnd_toEndOf="parent"
           app:layout_constraintHorizontal_bias="0.0"
           app:layout_constraintStart_toStartOf="parent"
-          app:layout_constraintTop_toBottomOf="@+id/input_pin"
-          profile:error="@{viewModel.confirmPinErrorMsg}" />
+          app:layout_constraintTop_toBottomOf="@+id/admin_pin_input_pin">
+
+          <com.google.android.material.textfield.TextInputEditText
```
Comment (FareesHussain) [inline app/src/main/res/layout-sw600dp-land/admin_pin_activity.xml:100]: Add this to support RTL for TextInputLayout
`android:textDirection="locale"`

File: app/src/main/res/layout-sw600dp-port/admin_pin_activity.xml [hunk 8]
```diff
@@ -65,35 +64,53 @@
...
+          app:errorMessage="@{viewModel.pinErrorMsg}"
           app:layout_constraintEnd_toEndOf="parent"
           app:layout_constraintStart_toStartOf="parent"
-          app:layout_constraintTop_toBottomOf="@+id/admin_pin_warning_text"
-          profile:error="@{viewModel.pinErrorMsg}" />
+          app:layout_constraintTop_toBottomOf="@+id/admin_pin_warning_text">
 
-        <org.oppia.android.app.profile.ProfileInputView
-          android:id="@+id/input_confirm_pin"
+          <com.google.android.material.textfield.TextInputEditText
```
Comment (FareesHussain) [inline app/src/main/res/layout-sw600dp-port/admin_pin_activity.xml:81]: Add this to support RTL for TextInputLayout
`android:textDirection="locale"`

File: app/src/main/res/layout-sw600dp-port/admin_pin_activity.xml [hunk 9]
```diff
@@ -65,35 +64,53 @@
...
-          app:isPasswordInput="true"
-          app:label="@string/admin_pin_new_confirm"
+          app:errorMessage="@{viewModel.confirmPinErrorMsg}"
           app:layout_constraintEnd_toEndOf="parent"
           app:layout_constraintStart_toStartOf="parent"
-          app:layout_constraintTop_toBottomOf="@+id/input_pin"
-          profile:error="@{viewModel.confirmPinErrorMsg}" />
+          app:layout_constraintTop_toBottomOf="@+id/admin_pin_input_pin">
+
+          <com.google.android.material.textfield.TextInputEditText
```
Comment (FareesHussain) [inline app/src/main/res/layout-sw600dp-port/admin_pin_activity.xml:107]: Add this to support RTL for TextInputLayout
`android:textDirection="locale"`

File: app/src/main/res/layout/admin_pin_activity.xml [hunk 10]
```diff
@@ -57,35 +56,53 @@
...
+        app:errorMessage="@{viewModel.pinErrorMsg}"
         app:layout_constraintEnd_toEndOf="parent"
         app:layout_constraintStart_toStartOf="parent"
-        app:layout_constraintTop_toBottomOf="@+id/admin_pin_warning_text"
-        profile:error="@{viewModel.pinErrorMsg}" />
+        app:layout_constraintTop_toBottomOf="@+id/admin_pin_warning_text">
 
-      <org.oppia.android.app.profile.ProfileInputView
-        android:id="@+id/input_confirm_pin"
+        <com.google.android.material.textfield.TextInputEditText
```
Comment (FareesHussain) [inline app/src/main/res/layout/admin_pin_activity.xml:73]: Add this to support RTL for TextInputLayout
`android:textDirection="locale"`

File: app/src/main/res/layout/admin_pin_activity.xml [hunk 11]
```diff
@@ -57,35 +56,53 @@
...
-        app:isPasswordInput="true"
-        app:label="@string/admin_pin_new_confirm"
+        app:errorMessage="@{viewModel.confirmPinErrorMsg}"
         app:layout_constraintEnd_toEndOf="parent"
         app:layout_constraintStart_toStartOf="parent"
-        app:layout_constraintTop_toBottomOf="@+id/input_pin"
-        profile:error="@{viewModel.confirmPinErrorMsg}" />
+        app:layout_constraintTop_toBottomOf="@+id/admin_pin_input_pin">
+
+        <com.google.android.material.textfield.TextInputEditText
```
Comment (FareesHussain) [inline app/src/main/res/layout/admin_pin_activity.xml:99]: Add this to support RTL for TextInputLayout
`android:textDirection="locale"`

**Review by FareesHussain — CHANGES_REQUESTED** [review]
@prayutsu NIT changes to avoid further issues related to RTL

**Comment by anandwana001** [comment]
@prayutsu Could you confirm test cases in `AdminPinActivityTest` are passing for robolectric?
```kotlin
testAdminAuthActivity_inputPin_inputConfirmPin_clickImeActionButton_checkOpensAdministratorControlsActivity

Error
androidx.test.espresso.base.DefaultFailureHandler$AssertionFailedWithCauseError: Wanted to match 1 intents. Actually matched 0 intents.

    IntentMatcher: has component: has component with: class name: is "org.oppia.android.app.administratorcontrols.AdministratorControlsActivity" package name: an instance of java.lang.String short class name: an instance of java.lang.String

    Matched intents:[]

    Recorded intents:[]
```

```kotlin
testAdminPinActivity_configurationChange_inputPin_inputWrongConfirmPin_clickImeActionButton_checkConfirmWrongError 

Error 
androidx.test.espresso.base.DefaultFailureHandler$AssertionFailedWithCauseError: 'TextInputLayout's error' doesn't match the selected view.
    Expected: TextInputLayout's error
         Got: "TextInputLayout{id=2131296357, res-name=admin_pin_input_confirm_pin, visibility=VISIBLE, width=792, height=178, has-focus=false, has-focusable=true, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=androidx.constraintlayout.widget.ConstraintLayout$LayoutParams@3dfc6b2, tag=null, root-is-layout-requested=false, has-input-connection=false, x=84.0, y=536.0, child-count=2}"
```

File: app/src/main/res/layout-land/admin_pin_activity.xml [hunk 12]
```diff
@@ -57,36 +56,54 @@
...
+        app:layout_constraintTop_toBottomOf="@+id/admin_pin_warning_text">
 
-      <org.oppia.android.app.profile.ProfileInputView
-        android:id="@+id/input_confirm_pin"
+        <com.google.android.material.textfield.TextInputEditText
+          android:id="@+id/admin_pin_input_pin_edit_text"
+          style="@style/TextInputEditText"
+          android:layout_width="match_parent"
+          android:layout_height="wrap_content"
+          android:hint="@string/admin_pin_new_pin"
```
Comment (anandwana001) [inline app/src/main/res/layout-land/admin_pin_activity.xml]: @prayutsu As per the official doc reference https://github.com/oppia/oppia-android/pull/2350#discussion_r550440503 we need to put hint in the `TextInputLayout `

File: app/src/main/res/layout-land/admin_pin_activity.xml (hunk 4 above)
Comment (prayutsu) [reply app/src/main/res/layout-land/admin_pin_activity.xml:77]: done

File: app/src/main/res/layout-land/admin_pin_activity.xml (hunk 5 above)
Comment (prayutsu) [reply app/src/main/res/layout-land/admin_pin_activity.xml:100]: done

File: app/src/main/res/layout-sw600dp-land/admin_pin_activity.xml (hunk 6 above)
Comment (prayutsu) [reply app/src/main/res/layout-sw600dp-land/admin_pin_activity.xml:75]: done

File: app/src/main/res/layout-sw600dp-land/admin_pin_activity.xml (hunk 7 above)
Comment (prayutsu) [reply app/src/main/res/layout-sw600dp-land/admin_pin_activity.xml:100]: done

File: app/src/main/res/layout-sw600dp-port/admin_pin_activity.xml (hunk 8 above)
Comment (prayutsu) [reply app/src/main/res/layout-sw600dp-port/admin_pin_activity.xml:81]: done

File: app/src/main/res/layout-sw600dp-port/admin_pin_activity.xml (hunk 9 above)
Comment (prayutsu) [reply app/src/main/res/layout-sw600dp-port/admin_pin_activity.xml:107]: done

File: app/src/main/res/layout/admin_pin_activity.xml (hunk 10 above)
Comment (prayutsu) [reply app/src/main/res/layout/admin_pin_activity.xml:73]: done

File: app/src/main/res/layout/admin_pin_activity.xml (hunk 11 above)
Comment (prayutsu) [reply app/src/main/res/layout/admin_pin_activity.xml:99]: done

File: app/src/main/res/layout-land/admin_pin_activity.xml (hunk 12 above)
Comment (prayutsu) [reply app/src/main/res/layout-land/admin_pin_activity.xml]: done

**Comment by prayutsu** [comment]
> @prayutsu Could you confirm test cases in `AdminPinActivityTest` are passing for robolectric?
> 
> ```kotlin
> testAdminAuthActivity_inputPin_inputConfirmPin_clickImeActionButton_checkOpensAdministratorControlsActivity
> 
> Error
> androidx.test.espresso.base.DefaultFailureHandler$AssertionFailedWithCauseError: Wanted to match 1 intents. Actually matched 0 intents.
> 
>     IntentMatcher: has component: has component with: class name: is "org.oppia.android.app.administratorcontrols.AdministratorControlsActivity" package name: an instance of java.lang.String short class name: an instance of java.lang.String
> 
>     Matched intents:[]
> 
>     Recorded intents:[]
> ```
> 
> ```kotlin
> testAdminPinActivity_configurationChange_inputPin_inputWrongConfirmPin_clickImeActionButton_checkConfirmWrongError 
> 
> Error 
> androidx.test.espresso.base.DefaultFailureHandler$AssertionFailedWithCauseError: 'TextInputLayout's error' doesn't match the selected view.
>     Expected: TextInputLayout's error
>          Got: "TextInputLayout{id=2131296357, res-name=admin_pin_input_confirm_pin, visibility=VISIBLE, width=792, height=178, has-focus=false, has-focusable=true, has-window-focus=true, is-clickable=false, is-enabled=true, is-focused=false, is-focusable=false, is-layout-requested=false, is-selected=false, layout-params=androidx.constraintlayout.widget.ConstraintLayout$LayoutParams@3dfc6b2, tag=null, root-is-layout-requested=false, has-input-connection=false, x=84.0, y=536.0, child-count=2}"
> ```

@anandwana001 All the tests are passing now, I am updating the screenshots for the same

File: .idea/gradle.xml [hunk 13]
```diff
@@ -23,4 +23,4 @@
       </GradleProjectSettings>
     </option>
   </component>
-</project>
+</project>
```
Comment (prayutsu) [inline .idea/gradle.xml]: This error seems irritating, I'll add this file to `.gitignore` but for now any suggestions to fix this?

Comment (FareesHussain) [reply .idea/gradle.xml]: @prayutsu you can use github and edit this file in your branch

File: app/src/main/java/org/oppia/android/app/profile/AdminPinActivityPresenter.kt [hunk 14]
```diff
@@ -48,43 +47,27 @@ class AdminPinActivityPresenter @Inject constructor(
...
 
-    binding.inputPin.post {
-      addTextChangedListener(binding.inputPin) { pin ->
-        pin?.let {
-          adminViewModel.pinErrorMsg.set("")
-          adminViewModel.savedPin.set(it.toString())
-          inputtedPin = pin.isNotEmpty()
-          setValidPin()
-        }
+    binding.adminPinInputPinEditText.onTextChanged { pin ->
```
Comment (FareesHussain) [inline app/src/main/java/org/oppia/android/app/profile/AdminPinActivityPresenter.kt:51]: Add a kdoc for this 
Ref - [kdoc for onTestChanged](https://github.com/anandwana001/oppia-android/blob/462f7fcab7637b0fe1083a254863cf32f528c60b/app/src/main/java/org/oppia/android/app/profile/AdminAuthActivityPresenter.kt#L49)

**Review by FareesHussain — CHANGES_REQUESTED** [review]
@prayutsu PTAL
I think kdoc is required for ext fun onTextChanged

File: app/src/main/java/org/oppia/android/app/profile/AdminPinActivityPresenter.kt [hunk 15]
```diff
@@ -48,43 +47,27 @@ class AdminPinActivityPresenter @Inject constructor(
...
 
-    binding.inputConfirmPin.post {
-      addTextChangedListener(binding.inputConfirmPin) { confirmPin ->
-        confirmPin?.let {
-          adminViewModel.confirmPinErrorMsg.set("")
-          adminViewModel.savedConfirmPin.set(it.toString())
-          inputtedConfirmPin = confirmPin.isNotEmpty()
-          setValidPin()
-        }
+    binding.adminPinInputConfirmPinEditText.onTextChanged { confirmPin ->
```
Comment (FareesHussain) [inline app/src/main/java/org/oppia/android/app/profile/AdminPinActivityPresenter.kt:61]: Add a kdoc for this 
Ref - [kdoc for onTestChanged](https://github.com/anandwana001/oppia-android/blob/462f7fcab7637b0fe1083a254863cf32f528c60b/app/src/main/java/org/oppia/android/app/profile/AdminAuthActivityPresenter.kt#L49)

File: .idea/gradle.xml (hunk 13 above)
Comment (prayutsu) [reply .idea/gradle.xml:27]: is it a good practice to edit this from here?

**Review by FareesHussain — APPROVED** [review]
LGTM, 
you can resolve all the comments i've added as it can only be resolved by a author or a member

File: app/src/main/java/org/oppia/android/app/profile/AdminPinActivityPresenter.kt (hunk 15 above)
Comment (prayutsu) [reply app/src/main/java/org/oppia/android/app/profile/AdminPinActivityPresenter.kt:61]: Done

File: app/src/main/java/org/oppia/android/app/profile/AdminPinActivityPresenter.kt (hunk 14 above)
Comment (prayutsu) [reply app/src/main/java/org/oppia/android/app/profile/AdminPinActivityPresenter.kt:51]: Done

File: app/src/main/java/org/oppia/android/app/profile/AdminPinActivityPresenter.kt [hunk 16]
```diff
@@ -139,20 +124,25 @@ class AdminPinActivityPresenter @Inject constructor(
...
+    binding.adminPinInputPinEditText.setOnEditorActionListener { _, actionId, event ->
+      if (actionId == EditorInfo.IME_ACTION_DONE ||
+        (event != null && (event.keyCode == KeyEvent.KEYCODE_ENTER))
+      ) {
+        binding.submitButton.callOnClick()
+      }
+      false
+    }
+    binding.adminPinInputPinEditText.setText(adminViewModel.savedPin.get().toString())
+    binding.adminPinInputConfirmPinEditText.setText(adminViewModel.savedConfirmPin.get().toString())
```
Comment (anandwana001) [inline app/src/main/java/org/oppia/android/app/profile/AdminPinActivityPresenter.kt]: earlier we use `setInput` function from `ProfileInputView` but after the migration can we perform text setting thing via data binding!!
@prayutsu @FareesHussain WDYT?

File: app/src/main/res/layout-land/admin_pin_activity.xml [hunk 17]
```diff
@@ -57,36 +56,56 @@
...
+        style="@style/TextInputLayout"
         android:layout_width="match_parent"
         android:layout_height="wrap_content"
         android:layout_marginStart="168dp"
         android:layout_marginTop="48dp"
         android:layout_marginEnd="168dp"
-        app:inputLength="5"
-        app:isPasswordInput="true"
-        app:label="@string/admin_pin_new_pin"
+        app:errorMessage="@{viewModel.pinErrorMsg}"
```
Comment (anandwana001) [inline app/src/main/res/layout-land/admin_pin_activity.xml:68]: reformat and rearrange, same for other xml file

File: app/src/sharedTest/java/org/oppia/android/app/profile/AdminPinActivityTest.kt [hunk 18]
```diff
@@ -92,6 +95,7 @@ import javax.inject.Singleton
   application = AdminPinActivityTest.TestApplication::class,
   qualifiers = "port-xxhdpi"
 )
+
```
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/profile/AdminPinActivityTest.kt]: empty line not needed.

File: app/src/sharedTest/java/org/oppia/android/app/profile/AdminPinActivityTest.kt [hunk 19]
```diff
@@ -285,29 +341,37 @@ class AdminPinActivityTest {
...
       )
       testCoroutineDispatchers.runCurrent()
       onView(withId(R.id.submit_button)).perform(nestedScrollTo()).perform(click())
-      onView(allOf(withId(R.id.error_text), isDescendantOfA(withId(R.id.input_confirm_pin)))).check(
+      onView(withId(R.id.admin_pin_input_confirm_pin)).check(
         matches(
-          withText(
-            context.getString(R.string.admin_pin_error_pin_confirm_wrong)
+          hasErrorText(
+            (R.string.admin_pin_error_pin_confirm_wrong)
```
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/profile/AdminPinActivityTest.kt]: Why do we need extra brackets?
```kotlin
hasErrorText(R.string.admin_pin_error_pin_confirm_wrong)
```

File: app/src/sharedTest/java/org/oppia/android/app/profile/AdminPinActivityTest.kt [hunk 20]
```diff
@@ -285,29 +341,37 @@ class AdminPinActivityTest {
...
+          isDescendantOfA(withId(R.id.admin_pin_input_confirm_pin))
+        )
+      ).perform(
         editTextInputAction.appendText("1234"),
         closeSoftKeyboard()
       )
       testCoroutineDispatchers.runCurrent()
       onView(withId(R.id.submit_button)).perform(nestedScrollTo()).perform(click())
-      onView(allOf(withId(R.id.error_text), isDescendantOfA(withId(R.id.input_confirm_pin)))).check(
+      onView(withId(R.id.admin_pin_input_confirm_pin)).check(
```
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/profile/AdminPinActivityTest.kt]: nit formatting

```kotlin
      onView(withId(R.id.admin_pin_input_confirm_pin))
        .check(matches(hasErrorText(R.string.admin_pin_error_pin_confirm_wrong)))
```

Comment (anandwana001) [reply app/src/sharedTest/java/org/oppia/android/app/profile/AdminPinActivityTest.kt]: applies for other too

**Review by anandwana001 — CHANGES_REQUESTED** [review]
Thanks @prayutsu 
nit changes suggested

File: app/src/main/java/org/oppia/android/app/profile/AdminPinActivityPresenter.kt (hunk 16 above)
Comment (FareesHussain) [reply app/src/main/java/org/oppia/android/app/profile/AdminPinActivityPresenter.kt:145]: @anandwana001 we have to do two-way data binding here??

Comment (prayutsu) [reply app/src/main/java/org/oppia/android/app/profile/AdminPinActivityPresenter.kt]: @anandwana001 Using data binding is a better practice, so we should definitely use it.

File: app/src/main/res/layout-land/admin_pin_activity.xml (hunk 17 above)
Comment (prayutsu) [reply app/src/main/res/layout-land/admin_pin_activity.xml:68]: Done

File: app/src/sharedTest/java/org/oppia/android/app/profile/AdminPinActivityTest.kt (hunk 18 above)
Comment (prayutsu) [reply app/src/sharedTest/java/org/oppia/android/app/profile/AdminPinActivityTest.kt]: Done

File: app/src/sharedTest/java/org/oppia/android/app/profile/AdminPinActivityTest.kt (hunk 19 above)
Comment (prayutsu) [reply app/src/sharedTest/java/org/oppia/android/app/profile/AdminPinActivityTest.kt]: Done

File: app/src/sharedTest/java/org/oppia/android/app/profile/AdminPinActivityTest.kt (hunk 20 above)
Comment (prayutsu) [reply app/src/sharedTest/java/org/oppia/android/app/profile/AdminPinActivityTest.kt]: Done

**Review by anandwana001 — APPROVED** [review]
Thanks @prayutsu 
Could you please do one more nit formatting at test file? Named arguments for the params pass while launching the test. 
```kotlin
context = context,
profileId = 0,
colorRgb = -10710042,
adminPinEnum = 1
```

Deferring to @FareesHussain to check Data binding.

**Comment by FareesHussain** [comment]
> 
> Deferring to @FareesHussain to check Data binding.

@anandwana001 What he used looks good to me.
As we just need to remove set input in Presenter.

**Comment by prayutsu** [comment]
> Thanks @prayutsu
> Could you please do one more nit formatting at test file? Named arguments for the params pass while launching the test.
> 
> ```kotlin
> context = context,
> profileId = 0,
> colorRgb = -10710042,
> adminPinEnum = 1
> ```
> 
> Deferring to @FareesHussain to check Data binding.

Done

**Review by rt4914 — APPROVED** [review]
LGTM, thanks.

**Pull request merged**